### PR TITLE
Allow overlay class choice in `SingleTargetAudioDataset`

### PR DESCRIPTION
**Pull request by rhine3**
Allow users to choose an overlay class by label.

When getting a sample with overlay, the default behavior is to overlay the sample with a sample from a different class. This is the desired behavior for multi-class, single-target training. (This would also be the desired behavior for multi-target training, but we would have to create a class that modifies the targets for each sample. `SingleTargetAudioDataset` is conceptualized as a single-target-only Dataset.)

But for single-species classifiers, where the two classes represent "present" and "absent," the desired behavior is to never overlay with files from the present class. Otherwise, the absent class will contain a high rate of true presences. With this PR, this behavior can be accomplished by setting the overlay class to the absent class label, e.g. `0` if using numeric labels.

### CODE REVIEW

File: opensoundscape/datasets.py [hunk 1]
```diff
@@ -219,6 +219,10 @@ class SingleTargetAudioDataset(torch.utils.data.Dataset):
         overlay_weight: The weight given to the overlaid image during augmentation.
             When 'random', will randomly select a different weight between 0.2 and 0.5 for each overlay
             When not 'random', should be a float between 0 and 1 [default: 'random']
+        overlay_class: The label of the class that overlays should be drawn from.
```
Comment (sammlapp) [inline opensoundscape/datasets.py:222]: the user should have to "get this right" by selecting the absence class label for binary model. As is, they could run with the default values and repeat our current bug.

Comment (rhine3) [reply opensoundscape/datasets.py:228]: Added error-checking to force user to "get it right" by either selecting absence class label or specifying the multi-class behavior with `overlay_class = 'different'`
